third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -163,6 +323,8 @@ struct LoadOpConversion : public ConvertOpToLLVMPattern<triton::LoadOp>,
         typeConverter->convertType(getElementTypeOrSelf(valueTy));
     unsigned vec = getVectorSize(ptr);
     unsigned numElems = getTotalElemsPerThread(ptr.getType());
+    bool useBufferOps =
```

Comment:
It's not good to have each invocation of the pattern to check environment variable. (In the extreme sense you can update the env while running the patterns so different invocations may see different value!) What about query it once in the pass `TritonGPUToLLVM.cpp` and control there?